crates/trie/trie/src/proof_v2/node.rs

```diff
@@ -16,15 +17,17 @@ pub(crate) enum ProofTrieBranchChild<RF> {
         /// The [`DeferredValueEncoder`] which will encode the leaf's value.
         value: RF,
     },
-    /// An extension node whose child branch has not yet been converted to an [`RlpNode`]
+    /// An extension node whose child branch has been converted to an [`RlpNode`]
     Extension {
         /// The short key of the leaf.
         short_key: Nibbles,
-        /// The node of the child branch.
-        child: BranchNode,
+        /// The [`RlpNode`] of the child branch.
+        child: RlpNode,
     },
     /// A branch node whose children have already been flattened into [`RlpNode`]s.
     Branch(BranchNode),
+    // A node whose type is not known, as it has already been converted to an [`RlpNode`].
```

Comment:
```suggestion
    /// A node whose type is not known, as it has already been converted to an [`RlpNode`].
```

smol nit for consistency